querydsl-jpa/src/test/java/com/querydsl/jpa/AbstractJPATest.java

```diff
@@ -319,6 +319,11 @@ public void Case3() {
     }
 
     @Test
+    public void Case4() {
+        query().from(cat).list(cat.mate.when(savedCats.get(0)).then(0).otherwise(1));
```

Comment:
You might want to assert something here.

Also, you added a test named `Case4` to #1305 .